memory_test.go

```diff
@@ -395,6 +395,15 @@ func TestMemCheck(t *testing.T) {
 			expectedSizeDiff: testNativeValueMemUsage +
 				(2 + SizeString), // "nv",
 		},
+		{
+			desc: "array.map function",
+			script: `x = [1,2,3,4,5]
+			x.map(_ => {
+				checkMem();
+				return "a"
+			})`,
+			expectedSizeDiff: 4 * (1 + SizeString),
```

Comment:
The notable thing here is that we are now seeing a size diff (this is 0 before this fix)